src/clients/MultiCallerClient.ts

```diff
@@ -261,12 +261,12 @@ export class MultiCallerClient {
   }
 
   buildMultiCallBundle(transactions: AugmentedTransaction[]): AugmentedTransaction {
-    // Validate all transactions in the batch have the same target contract.
-    const target = transactions[0].contract;
-    if (transactions.every((tx) => tx.contract.address !== target.address)) {
+    // Validate all transactions have the same chainId and target contract.
+    const { chainId, contract } = transactions[0];
+    if (transactions.some((tx) => tx.contract.address !== contract.address || tx.chainId !== chainId)) {
```

Comment:
Maybe this works too? not sure if its more efficient though?
```suggestion
    if (transactions.slice(1).some((tx) => tx.contract.address !== contract.address || tx.chainId !== chainId)) {
```

Reply:
Ah yeah - it would be nice to skip over the first element, since we know it will always be false in this test.

I checked on the documentation for Array.slice(). It returns an entirely new object with a shallow copy of a subset of the input array. I suspect that it would probably cost more than just wasting a loop on the first element, and it would also cost more cycles as the length of transactions increases (i.e. there are just additional references to create in the new array).

It does annoy me that we make a redundant check on the first element, though :-1:. It'd be really nice to just specify an offset into the array without creating an entirely new object...as far as I can tell, Javascript doesn't allow us to do that :(